### DISCUSSION TITLE
How can I use only OSS in Pytorch lightning?

### DISCUSSION THREAD

**Discussion by CD21a** [Lightning-AI/pytorch-lightning#14105] [DDP / multi-GPU / multi-node]
Hello

I wanna use only OSS (zero1) in pytorch lightning, but pytorch lightning only support OSS + SDP (zero2).
How can i use only OSS?

**Comment by awaelchli** [comment]
Hi
You can do optimizer state sharding with [deepspeed stage 1 (zero1)](https://pytorch-lightning.readthedocs.io/en/stable/advanced/model_parallel.html#deepspeed-zero-stage-1
):

```py
trainer = Trainer(accelerator="gpu", devices=4, strategy="deepspeed_stage_1", precision=16)
```

**Reply by awaelchli** [reply]
If you don't want to use deepspeed, you can also just use the OSS optimizer in PyTorch: https://pytorch.org/tutorials/recipes/zero_redundancy_optimizer.html
